homeassistant/components/bluetooth_tracker/device_tracker.py

```diff
@@ -42,66 +44,91 @@
+        device_id=device_id,
+    )
+    _LOGGER.debug("Bluetooth devices discovered = %d", len(result))
+    return result
+
+
+def see_device(see, mac: str, device_name: str, rssi=None) -> None:
+    """Mark a device as seen."""
+    attributes = {}
+    if rssi is not None:
+        attributes["rssi"] = rssi
+    see(
+        mac="{}{}".format(BT_PREFIX, mac),
+        host_name=device_name,
+        attributes=attributes,
+        source_type=SOURCE_TYPE_BLUETOOTH,
+    )
+
+
+def get_tracking_devices(hass: HomeAssistantType) -> Tuple[Set[str], Set[str]]:
+    """
+    Load all known devices.
+
+    We just need the devices so set consider_home and home range to 0
+    """
+    yaml_path: str = hass.config.path(YAML_DEVICES)
+    devices_to_track: Set[str] = set()
+    devices_to_not_track: Set[str] = set()
+
     for device in run_coroutine_threadsafe(
         async_load_config(yaml_path, hass, 0), hass.loop
     ).result():
         # Check if device is a valid bluetooth device
-        if device.mac and device.mac[:3].upper() == BT_PREFIX:
-            if device.track:
-                devs_to_track.append(device.mac[3:])
-            else:
-                devs_donot_track.append(device.mac[3:])
+        if not is_bluetooth_device(device):
+            continue
+
+        normalized_mac: str = device.mac[3:]
+        if device.track:
+            devices_to_track.add(normalized_mac)
+        else:
+            devices_to_not_track.add(normalized_mac)
+
+    return devices_to_track, devices_to_not_track
+
+
+def setup_scanner(hass: HomeAssistantType, config: dict, see, discovery_info=None):
+    """Set up the Bluetooth Scanner."""
+    # pylint: disable=import-error
+    import bluetooth
+
+    device_id: int = config.get(CONF_DEVICE_ID)
+    devices_to_track, devices_to_not_track = get_tracking_devices(hass)
 
     # If track new devices is true discover new devices on startup.
-    track_new = config.get(CONF_TRACK_NEW, DEFAULT_TRACK_NEW)
+    track_new: bool = config.get(CONF_TRACK_NEW, DEFAULT_TRACK_NEW)
+    _LOGGER.debug("Tracking new devices = %s", track_new)
+
+    if not devices_to_track and not track_new:
+        _LOGGER.warning("No Bluetooth devices to track and not tracking new devices")
+        return False
+
     if track_new:
-        for dev in discover_devices():
-            if dev[0] not in devs_to_track and dev[0] not in devs_donot_track:
-                devs_to_track.append(dev[0])
-                see_device(dev[0], dev[1])
+        for mac, device_name in discover_devices(bluetooth, device_id):
+            if mac not in devices_to_track and mac not in devices_to_not_track:
+                devices_to_track.add(mac)
+                see_device(see, mac, device_name)
 
     interval = config.get(CONF_SCAN_INTERVAL, SCAN_INTERVAL)
 
     request_rssi = config.get(CONF_REQUEST_RSSI, False)
+    if request_rssi:
+        _LOGGER.debug("Detecting RSSI for devices")
+        from bt_proximity import BluetoothRSSI
```

Comment:
What's the benefit of importing this on demand?

Reply:
I haven't measured it, but seen it done in other modules. Is it redundant? This modulr is only loaded for a subset of users..

Reply:
Yes, home assistant loads integration specific requirements on demand by default.